tests/src/tests/membershipCreationTest.ts

```diff
@@ -0,0 +1,92 @@
+import { WsProvider } from '@polkadot/api';
+import { registerJoystreamTypes } from '@joystream/types';
+import { Keyring } from '@polkadot/keyring';
+import { assert } from 'chai';
+import { KeyringPair } from '@polkadot/keyring/types';
+import BN = require('bn.js');
+import { ApiMethods } from '../utils/apiMethods';
+import { initConfig } from '../utils/config';
+
+describe('Membership integration tests', () => {
+  initConfig();
+  const keyring = new Keyring({ type: 'sr25519' });
+  const nKeyPairs: KeyringPair[] = new Array();
+  const N: number = +process.env.MEMBERSHIP_CREATION_N!;
+  const paidTerms: number = +process.env.MEMBERSHIP_PAID_TERMS!;
+  const nodeUrl: string = process.env.NODE_URL!;
+  const sudoUri: string = process.env.SUDO_ACCOUNT_URL!;
+  const defaultTimeout: number = 30000;
+  let apiMethods: ApiMethods;
+  let sudo: KeyringPair;
+  let aKeyPair: KeyringPair;
+  let membershipFee: number;
+  let membershipTransactionFee: number;
+
+  before(async function () {
+    this.timeout(defaultTimeout);
+    registerJoystreamTypes();
+    const provider = new WsProvider(nodeUrl);
+    apiMethods = await ApiMethods.create(provider);
+    sudo = keyring.addFromUri(sudoUri);
+    for (let i = 0; i < N; i++) {
+      nKeyPairs.push(keyring.addFromUri(i.toString()));
+    }
+    aKeyPair = keyring.addFromUri('A');
+    membershipFee = await apiMethods.getMembershipFee(paidTerms);
+    membershipTransactionFee = apiMethods.estimateBuyMembershipFee(
+      sudo,
+      paidTerms,
+      'member_name_which_is_longer_than_expected'
+    );
+    let nonce = await apiMethods.getNonce(sudo);
+    nonce = nonce.sub(new BN(1));
```

Comment:
Subtraction was required because of nonce incremention happening in the loop before sending the transaction. Now fixed with introduction of the nonce management.